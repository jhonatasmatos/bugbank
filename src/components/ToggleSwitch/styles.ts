import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
