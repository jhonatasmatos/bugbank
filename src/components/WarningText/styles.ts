import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
