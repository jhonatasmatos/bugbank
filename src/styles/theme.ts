export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
