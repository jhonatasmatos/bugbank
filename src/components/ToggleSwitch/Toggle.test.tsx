import { fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import { ToggleSwitch } from '.';

const fn = jest.fn();

describe('ToggleSwitch', () => {
  it('should render background with gray color when isChecked is false', () => {
    const { container } = renderTheme(
      <ToggleSwitch id="toggleId" isChecked={false} onClick={fn} />,
    );

    const containerToggle = container.querySelector('label');
    const span = container.querySelector('span');

    expect(containerToggle).toHaveStyle(`background: ${theme.colors.gray};`);
    expect(span).toHaveStyle('left: 0;');
  });

  it('should render background with primary color when isChecked is true', () => {
    const { container } = renderTheme(
      <ToggleSwitch id="toggleId" isChecked={true} onClick={fn} />,
    );

    const containerToggle = container.querySelector('label');
    const inputToggle = container.querySelector('input');
    const span = container.querySelector('span');

    fireEvent.click(inputToggle);

    expect(fn).toHaveBeenCalledTimes(1);
    expect(containerToggle).toHaveStyle(`background: ${theme.colors.primary};`);
    expect(span).toHaveStyle('left: 60%;');
  });

  it('should render with checked false', () => {
    const { container } = renderTheme(
      <ToggleSwitch id="toggleId" isChecked={false} onClick={fn} />,
    );
    expect(container).toMatchSnapshot();
  });
});
