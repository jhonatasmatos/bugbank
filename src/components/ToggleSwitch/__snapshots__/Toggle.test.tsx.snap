// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch should render with checked false 1`] = `
.c0 {
  border: 1px solid #D62A92;
  width: 5rem;
  height: 2rem;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  position: relative;
  background: #d3d3d3;
}

.c1 {
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

.c2 {
  position: absolute;
  width: 2rem;
  height: 2rem;
  border-radius: 1rem;
  background: #fff;
  left: 0;
  cursor: pointer;
  box-shadow: 2px 2px 4px #D62A92;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

<div>
  <label
    class="c0"
  >
    <input
      class="c1"
      id="toggleId"
    />
    <span
      class="c2"
    />
  </label>
</div>
`;
